Modules/_ctypes/stgdict.c

```diff
@@ -40,6 +40,7 @@ PyCStgInfo_clone(StgInfo *dst_info, StgInfo *src_info)
     Py_XINCREF(dst_info->converters);
     Py_XINCREF(dst_info->restype);
     Py_XINCREF(dst_info->checker);
+    Py_XINCREF(dst_info->pointer_type);
```

Comment:
`comtypes` should pass `test_safearray` with this:

```suggestion
    dst_info->pointer_type = NULL;  // the cache cannot be shared
```

Reply:
Yeah, it seems like you are right, thanks! I will add test.